tests/ui/asm_syntax.rs

```diff
@@ -0,0 +1,29 @@
+#![feature(asm)]
```

Comment:
```suggestion
// only-x86_64

#![feature(asm)]
```

With the `cfg` approach, the tests will fail if the architecture is not x86/x86_64 as the output will not match the `stderr` file.

The proposed approach is what seems to be used in most of rustc's `asm!()` tests. We can then remove all the `cfg` attributes.